node-graph/gcore/src/graphic_element/renderer.rs

```diff
@@ -670,6 +728,18 @@ impl GraphicElementRendered for VectorDataTable {
 
 	fn add_upstream_click_targets(&self, click_targets: &mut Vec<ClickTarget>) {
 		for instance in self.instance_ref_iter() {
+			// For free-floating anchors, we need to add a click target for each
+			let single_anchors = instance.instance.point_domain.ids().iter().filter(|&&point_id| instance.instance.connected_count(point_id) == 0);
+			let single_anchors_targets = single_anchors
+				.map(|&point_id| {
+					let anchor = instance.instance.point_domain.position_from_id(point_id).unwrap_or_default();
+					let group = ManipulatorGroup::new_anchor_with_id(anchor, point_id);
+					let mut click_target = ClickTarget::new_with_point_group(group);
+					click_target.apply_transform(*instance.transform);
+					click_target
+				})
+				.collect::<Vec<ClickTarget>>();
```

Comment:
There is no need to allocate if you are just going to collect. Also why is this created at the top and used at the bottom of the function?